packages/2d/src/components/CodeBlock.ts

```diff
@@ -47,14 +50,44 @@ export interface CodeModification {
 }
 
 export class CodeBlock extends Shape {
-  private static initialized = createComputedAsync(
-    () => ready().then(() => true),
-    false,
-  );
-
   @initial('tsx')
   @signal()
-  public declare readonly language: SimpleSignal<string, this>;
+  public declare readonly language: SimpleSignal<Lang, this>;
+
+  @initial(undefined)
+  @signal()
+  public declare readonly stockTheme: SimpleSignal<Theme | undefined, this>;
+
+  protected static loadedLanguages = new Map<Lang, boolean>();
+  protected static loadedThemes = new Map<Theme, boolean>();
```

Comment:
Looks like these could be simple Sets

Reply:
I actually intended for these to use `false` as well. The false means that the language or theme is loading but isn't loaded yet.